kong/plugins/opentelemetry/handler.lua

```diff
@@ -42,14 +46,24 @@ end
 
 
 function OpenTelemetryHandler:log(conf)
+  -- Read resource attributes variable
+  local options = {}
+  if conf.resource_attributes then
```

Comment:
In https://github.com/Kong/kong/blob/d5a0c2342407454fc78d35c2d7ce0e2ec3429968/kong/plugins/opentelemetry/schema.lua#L4-L20 we have vtype and value validator, does that mean even if the template variable returns a non-string (for example a number), its vtype has to be string and we would encode it as string?

Reply:
in 
https://github.com/Kong/kong/blob/d5a0c2342407454fc78d35c2d7ce0e2ec3429968/kong/plugins/opentelemetry/schema.lua#L22-L29
The resource value type have been restricted to be string. Not really sure about the plan of supporting numbers and boolean.

Reply:
good catch @fffonion , I am not sure why the constraint of only allowing strings currently exists, [the todo](https://github.com/Kong/kong/blob/88ffdda5904d0b375e973014951b65a107f593c5/kong/plugins/opentelemetry/schema.lua#L26) clearly shows the intention of supporting all value types in the future, but maybe it'd be safer to maintain the current behavior even for rendered attributes and convert to string before returning them, while we look into it and possibly update the schema to allow different types in the future.

Reply:
so actually I checked and this is already fine, because render [always returns a string](https://github.com/lunarmodules/Penlight/blob/d2dc61bd8f39e3686bf97169f7aa2ef59c8835e2/lua/pl/template.lua#L142).
`$(2 + 2)` is rendered as "4" (of type string), so the resource_attribute type would remain of type string after this change regardless of the source variable in the template.